samples/powerpoint/text/get-set-textrange.yaml

```diff
@@ -0,0 +1,191 @@
+order: 1
+id: powerpoint-text-get-set-textrange
+name: Work with text range selections
+description: 'Get, set, load, and save text range selections.'
+host: POWERPOINT
+api_set:
+    PowerPointApi: '1.5'
+script:
+    content: |
+        $("#getSelectedTextRange").click(() => tryCatch(getSelectedTextRange));
+        $("#setSelectedTextRange").click(() => tryCatch(setSelectedTextRange));
+        $("#changeColor").click(() => tryCatch(changeColor));
+        $("#saveTextSelection").click(() => tryCatch(saveTextSelection));
+        $("#loadTextSelection").click(() => tryCatch(loadTextSelection));
+
+        async function getSelectedTextRange() {
+          // Gets the selected text range and prints data about the range on the task pane.
+          await PowerPoint.run(async (context) => {
+            const textRange = context.presentation.getSelectedTextRange();
+            try {
+              await context.sync();
+            } catch (error) {
+              console.warn("You must select only one range of text for this action to work.");
+              return;
+            }
+            textRange.load("text");
+            textRange.load("start");
+            textRange.load("length");
+            await context.sync();
+            let txtHtml = textRange.text;
+            txtHtml = txtHtml.replace(/\n/g, "<br>");
+            txtHtml = txtHtml.replace(/\r/g, "<br>");
+            txtHtml = txtHtml.replace(/\v/g, "<br>");
+            let txtExplained = textRange.text;
+            txtExplained = txtExplained.replace(/\n/g, "<font color=red>NL</font>");
+            txtExplained = txtExplained.replace(/\r/g, "<font color=red>CR</font>");
+            txtExplained = txtExplained.replace(/\v/g, "<font color=red>VV</font>");
+            let finalTable = "";
+            finalTable +=
+              "<br><table border=1 cellpadding=3 cellspacing=0><tr><td bgcolor=#3333EE><font color=white>Index</font></td><td bgcolor=#3333EE><font color=white>Id</font></td></tr>";
+            finalTable += "<tr><td>Raw</td><td>" + textRange.text + "</td></tr>";
+            finalTable += "<tr><td>Html</td><td>" + txtHtml + "</td></tr>";
+            finalTable += "<tr><td>Exp</td><td>" + txtExplained + "</td></tr>";
+            finalTable += "<tr><td>Start</td><td>" + textRange.start + "</td></tr>";
+            finalTable += "<tr><td>Length</td><td>" + textRange.length + "</td></tr>";
+            finalTable += "</table>";
+            $("#outputSpan").empty();
+            $("#outputSpan").append(finalTable);
+          });
+        }
+        async function setSelectedTextRange() {
+          // Selects the first 10 characters of the selected shape.
```

Comment:
This method throws an error if the text in the shape has fewer than ten letters. I think confusion could be averted by updating the directions in the HTML section: 

"This sample shows how to get selected text, and how to select specific text. Select a shape that contains more than 10 letters to try out this sample."

Reply:
I added code to warn on this and updated the text.